_posts/2021-02-11-tabnet/tabnet.Rmd

```diff
@@ -0,0 +1,411 @@
+
+Certainly, it would be interesting to double-check these results on `tabnet`, and then, look at the respective feature importances. However, given the size of the dataset, non-negligible computing resources (and patience) will be required.
+
+Speaking of size, let's take a look:
+
+```{r}
+higgs %>% glimpse()
+```
+
+    Rows: 11,000,000
+    Columns: 29
+    $ class                    <fct> 1.000000000000000000e+00, 1.000000…
+    $ lepton_pT                <dbl> 0.8692932, 0.9075421, 0.7988347, 1…
+    $ lepton_eta               <dbl> -0.6350818, 0.3291473, 1.4706388, …
+    $ lepton_phi               <dbl> 0.225690261, 0.359411865, -1.63597…
+    $ missing_energy_magnitude <dbl> 0.3274701, 1.4979699, 0.4537732, 1…
+    $ missing_energy_phi       <dbl> -0.68999320, -0.31300953, 0.425629…
+    $ jet_1_pt                 <dbl> 0.7542022, 1.0955306, 1.1048746, 1…
+    $ jet_1_eta                <dbl> -0.24857314, -0.55752492, 1.282322…
+    $ jet_1_phi                <dbl> -1.09206390, -1.58822978, 1.381664…
+    $ jet_1_b_tag              <dbl> 0.000000, 2.173076, 0.000000, 0.00…
+    $ jet_2_pt                 <dbl> 1.3749921, 0.8125812, 0.8517372, 2…
+    $ jet_2_eta                <dbl> -0.6536742, -0.2136419, 1.5406590,…
+    $ jet_2_phi                <dbl> 0.9303491, 1.2710146, -0.8196895, …
+    $ jet_2_b_tag              <dbl> 1.107436, 2.214872, 2.214872, 2.21…
+    $ jet_3_pt                 <dbl> 1.1389043, 0.4999940, 0.9934899, 1…
+    $ jet_3_eta                <dbl> -1.578198314, -1.261431813, 0.3560…
+    $ jet_3_phi                <dbl> -1.04698539, 0.73215616, -0.208777…
+    $ jet_3_b_tag              <dbl> 0.000000, 0.000000, 2.548224, 0.00…
+    $ jet_4_pt                 <dbl> 0.6579295, 0.3987009, 1.2569546, 0…
+    $ jet_4_eta                <dbl> -0.01045457, -1.13893008, 1.128847…
+    $ jet_4_phi                <dbl> -0.0457671694, -0.0008191102, 0.90…
+    $ jet_4_btag               <dbl> 3.101961, 0.000000, 0.000000, 0.00…
+    $ m_jj                     <dbl> 1.3537600, 0.3022199, 0.9097533, 0…
+    $ m_jjj                    <dbl> 0.9795631, 0.8330482, 1.1083305, 1…
+    $ m_lv                     <dbl> 0.9780762, 0.9856997, 0.9856922, 0…
+    $ m_jlv                    <dbl> 0.9200048, 0.9780984, 0.9513313, 0…
+    $ m_bb                     <dbl> 0.7216575, 0.7797322, 0.8032515, 0…
+    $ m_wbb                    <dbl> 0.9887509, 0.9923558, 0.8659244, 1…
+    $ m_wwbb                   <dbl> 0.8766783, 0.7983426, 0.7801176, 0…
+
+11,000,000 "observations" (kind of) -- that's a lot! Like the authors of the TabNet paper (@arik2020tabnet), we'll use 500,000 of these for validation. (Unlike them, though, we won't be able to train for 870,000 iterations!)
```

Comment:
Numbers are always spelled out [when they begin a sentence](https://www.grammarbook.com/numbers/numbers.asp).

Reply:
```suggestion
Eleven million "observations" (kind of) -- that's a lot! Like the authors of the TabNet paper (@arik2020tabnet), we'll use 500,000 of these for validation. (Unlike them, though, we won't be able to train for 870,000 iterations!)
```

Reply:
good to know!